src/main.c

```diff
@@ -120,6 +133,29 @@ int handle_command_line_arguments(int argc, char **argv) {
         print_acceptable_calling_conventions();
         return 1;
       }
+    } else if (strcmp(argument, "-d") == 0
+               || strcmp(argument, "--dialect") == 0) {
+      i++;
+      if (i >= argc) {
+        panic("ERROR: Expected assembly dialect after format command line argument");
+      }
+      if (*argv[i] == '-') {
+        panic("ERROR: Expected assembly dialect after format command line argument\n"
+              "Instead, got what looks like another command line argument.\n"
+              " -> \"%s\"", argv[i]);
+      }
+      if (strcmp(argv[i], "default") == 0) {
+        output_assembly_dialect = CG_ASM_DIALECT_DEFAULT;
+      } else if (strcmp(argv[i], "att") == 0) {
+        output_assembly_dialect = CG_ASM_DIALECT_ATT;
+      } else if (strcmp(argv[i], "intel") == 0) {
+        output_assembly_dialect = CG_ASM_DIALECT_INTEL;
+      } else {
+        printf("ERROR: Expected assembly dialect after calling convention command line argument\n"
+               "Instead, got an unrecognized format: \"%s\".\n", argv[i]);
+        print_acceptable_asm_dialects();
+        return 1;
+      }
```

Comment:
GAS != dialect. Because you can have GAS+AT&T, GAS+Intel, GAS+NASM etc.

However, I agree that when you select the target architecture (x86_64), should shouldn't have to care what assembler is gonna be used, so maybe dialect, architecture, and assembler should all be separate options, or maybe we should have a GAS+AT&T dialect after all?